modules/team_routing_rule/main.tf

```diff
@@ -1,16 +1,48 @@
 resource "opsgenie_team_routing_rule" "this" {
   count = module.this.enabled ? 1 : 0
 
-  name     = var.team_routing_rule.name
-  team_id  = var.team_routing_rule.team_id
-  order    = try(var.team_routing_rule.order, 0)
+  name    = var.team_routing_rule.name
+  team_id = var.team_routing_rule.team_id
+  order   = try(var.team_routing_rule.order, 5)
+
   timezone = try(var.team_routing_rule.timezone, "America/Los_Angeles")
+  dynamic "time_restriction" {
+    for_each = [for time_restriction in try([var.team_routing_rule.time_restriction], []) : time_restriction if time_restriction != null]
+
+    content {
+      # NOTE: The Opsgenie Provider appears to be inconsistent with how it uses time_restriction:
+      # `restrictions` for type `weekday-and-time-of-day`
+      # `restriction` for type `time-of-day`
+      type = try(var.team_routing_rule.time_restriction.type, "weekday-and-time-of-day")
+      dynamic "restrictions" {
+        for_each = var.team_routing_rule.time_restriction.type == "weekday-and-time-of-day" ? try(var.team_routing_rule.time_restriction.restrictions, []) : []
+        content {
+          start_hour = try(restrictions.value.start_hour, 09)
+          start_min  = try(restrictions.value.start_min, 0)
+          start_day  = try(restrictions.value.start_day, "monday")
+          end_hour   = try(restrictions.value.end_hour, 17)
+          end_min    = try(restrictions.value.end_min, 00)
+          end_day    = try(restrictions.value.end_day, "friday")
+        }
+      }
+
+      dynamic "restriction" {
+        for_each = var.team_routing_rule.time_restriction.type == "time-of-day" ? try(var.team_routing_rule.time_restriction.restrictions, []) : []
```

Comment:
this expression means that type could be provided but the rest of the fields not.

Reply:
Correct, that was my initial thoughts, I think you're right though - we make all fields required if `var.team_routing_rule.time_restriction` is not null (line 10).

meaning if you are providing a time_restriction, then you must fully configure it.

This line separates the restrictions based on type. because of the GH issue